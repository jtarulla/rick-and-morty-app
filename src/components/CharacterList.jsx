import React from 'react';
import CharacterCard from 'components/CharacterCard';

const CharacterList = ({loading, characters}) => {
    return (
        <div className="character-card-list">
            <div className="container">
                {characters.map((character) => (
                    <CharacterCard
                        loading={loading}
                        key={character.id}
                        id={character.id}
                        name={character.name}
                        image={character.image}
                    />
                ))}
            </div>
        </div>
    );
}

export default CharacterList;