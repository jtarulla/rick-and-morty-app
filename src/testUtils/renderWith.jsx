import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
    return (
        <MemoryRouter initialEntries={initialRoutes}>
            {children}
        </MemoryRouter>
    );
};

export const renderWithRouter = (ui, options) => {
    const initialRoutes =
        options && options.initialRoutes ? options.initialRoutes : ["/"];

    return render(ui, {
            wrapper: (args) =>
                MemoryRouterWithInitialRoutes({
                    ...args,
                    initialRoutes,
                }),
                ...options,
        }
    );
}

export * from "@testing-library/react";
