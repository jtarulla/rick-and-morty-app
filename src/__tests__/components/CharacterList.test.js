import '@testing-library/jest-dom/extend-expect';
import { renderWithRouter } from 'testUtils/renderWith';

import CharacterList from 'components/CharacterList';
import { characterList, loading } from 'testUtils/mocks/characters'

const setup = async (isLoading = loading.false) => {
    const wrapper = await renderWithRouter(
        <CharacterList
            characters={characterList}
            loading={isLoading}
        />
    );

    return {wrapper};
}

describe('CharacterList component', () => {
    it('renders', async () => {
        const { input } = await setup();

        expect(input).toMatchSnapshot();
    });

    it('renders Card child component', async () => {
        const { wrapper } = await setup();
        const childComponents = await wrapper.findAllByTestId('character-card');

        expect(childComponents.length).toEqual(3);
    });

    it('renders Loading component if loading is true', async () => {
        const { wrapper } = await setup(loading.true);
        const loadingComponent = wrapper.getAllByTitle('Loading...', {exact: false});

        expect(loadingComponent).toBeTruthy();
    });
})