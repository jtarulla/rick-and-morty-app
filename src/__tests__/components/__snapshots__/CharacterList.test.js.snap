// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders 1`] = `undefined`;
